emmet-core/emmet/core/vasp/validation/check_incar.py

```diff
@@ -0,0 +1,770 @@
+    else:
+        warnings.append(f"SIGMA is not being directly checked, as an ISMEAR of {cur_ismear} is being used. However, given the bandgap of {round(bandgap,3)}, the maximum SIGMA used should be {valid_sigma} if using an ISMEAR *not* in [-5, -4, -2].")
+    
+    # Also check if SIGMA is too large according to the VASP wiki,
+    # which occurs when the entropy term in the energy is greater than 1 meV/atom.
+    all_eentropies_per_atom = []
+    for ionic_step in ionic_steps:
+        electronic_steps = ionic_step['electronic_steps']
+        # print(electronic_steps)
+        for elec_step in electronic_steps:
+            if 'eentropy' in elec_step.keys():
+                if elec_step['eentropy'] != None:
+                    all_eentropies_per_atom.append(elec_step['eentropy'] / structure.num_sites)
+    
+    cur_max_eentropy_per_atom = max(abs(np.array(all_eentropies_per_atom)))
+    valid_max_eentropy_per_atom = 0.001
+        
+    if cur_max_eentropy_per_atom > valid_max_eentropy_per_atom:
+        reasons.append(f"INPUT SETTINGS --> SIGMA: The entropy term (T*S) in the energy was {round(1000 * cur_max_eentropy_per_atom, 3)} meV/atom, which is "\
+                       f"greater than the {round(1000 * valid_max_eentropy_per_atom, 1)} meV/atom maximum suggested in the VASP wiki. "\
+                       f"Thus, SIGMA should be decreased.")
+
+
+def _check_lmaxmix_and_lmaxtau(reasons, warnings, parameters, incar, valid_input_set, structure, task_type):
+    """
+    Check that LMAXMIX and LMAXTAU are above the required value. Also ensure that they are not greater than 6, 
+    as that is inadvisable according to the VASP development team (as of writing this in August 2023).
+    """
+    
+    valid_lmaxmix = valid_input_set.incar.get("LMAXMIX", 2)
+    valid_lmaxtau = min(valid_lmaxmix + 2, 6)
+    lmaxmix_or_lmaxtau_too_high_msg = "From empirical testing, using LMAXMIX and / or LMAXTAU > 6 appears to introduce computational instabilities, " \
+        "and is currently inadvisable according to the VASP development team."
+    
+    # Write out checks manually to better control error message.
+    # LMAXMIX.
+    cur_lmaxmix = parameters.get("LMAXMIX", 2)
+    if (cur_lmaxmix < valid_lmaxmix) or (cur_lmaxmix > 6):
+        if valid_lmaxmix < 6:
+            lmaxmix_msg = f"INPUT SETTINGS --> LMAXMIX: value is set to {cur_lmaxmix}, but should be between {valid_lmaxmix} and 6."
+        else:
+            lmaxmix_msg = f"INPUT SETTINGS --> LMAXMIX: value is set to {cur_lmaxmix}, but should be {valid_lmaxmix}."
+        # add additional context for invalidation if user set LMAXMIX > 6
+        if cur_lmaxmix > 6:
+            lmaxmix_msg += lmaxmix_or_lmaxtau_too_high_msg
+            
+        # Either add to reasons or warnings depending on task type (as this affects NSCF calcs the most)
+        # @ Andrew Rosen, is this an adequate check? Or should we somehow also be checking for cases where
```

Comment:
@Andrew-S-Rosen thoughts on this check either:
1) creating a reason (which determines whether a calc is valid or not) or 
2) creating a warning 

simply depending on task type?